daemon/graphdriver/lcow/remotefs_filedriver.go

```diff
@@ -0,0 +1,248 @@
+// +build windows
+
+package lcow
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"bytes"
+
+	"io/ioutil"
+
+	"github.com/containerd/continuity/driver"
+	"github.com/sirupsen/logrus"
+)
+
+// TODO: @gupta-ak. These constants are defined in the Microsoft/opengcs repo.
+// Once docker switches to the using that repo for opengcs, we can remove these.
+const remotefsCmd = "remotefs"
+
+const (
+	statCmd           = "stat"
+	lstatCmd          = "lstat"
+	readlinkCmd       = "readlink"
+	mkdirCmd          = "mkdir"
+	mkdirAllCmd       = "mkdirall"
+	removeCmd         = "remove"
+	removeAllCmd      = "removeall"
+	linkCmd           = "link"
+	symlinkCmd        = "symlink"
+	lchmodCmd         = "lchmod"
+	lchownCmd         = "lchown"
+	mknodCmd          = "mknod"
+	mkfifoCmd         = "mkfifo"
+	openFileCmd       = "openfile"
+	readFileCmd       = "readfile"
+	writeFileCmd      = "writefile"
+	readDirCmd        = "readdir"
+	resolvePathCmd    = "resolvepath"
+	extractArchiveCmd = "extractarchive"
+	archivePathCmd    = "archivepath"
+)
+
+var _ driver.Driver = &lcowfs{}
+
+func (l *lcowfs) Open(path string) (driver.File, error) {
+	return l.OpenFile(path, os.O_RDONLY, 0)
+}
+
+func (l *lcowfs) OpenFile(path string, flag int, perm os.FileMode) (_ driver.File, err error) {
+	flagStr := strconv.FormatInt(int64(flag), 10)
+	permStr := strconv.FormatUint(uint64(perm), 8)
+
+	cmd := fmt.Sprintf("%s %s %s %s %s", remotefsCmd, openFileCmd, path, flagStr, permStr)
+	if err := l.runProcess(cmd, nil, nil, nil); err != nil {
+		return nil, err
+	}
+
+	// Assume that we want to follow symlinks.
+	fi, err := l.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// After OpenFile, we know that the file exists in the VM.
+	// If it's a regular file, retrieve it
+	var file *os.File
+	if !fi.IsDir() {
+		file, err = ioutil.TempFile("", "remotefs")
+		if err != nil {
+			return nil, err
+		}
+		fileName := file.Name()
+
+		cmd = fmt.Sprintf("%s %s %s", remotefsCmd, readFileCmd, path)
+		if err := l.runProcess(cmd, nil, file, nil); err != nil {
+			file.Close()
+			os.RemoveAll(fileName)
+			return nil, err
+		}
+
+		// We do this so that we can match the flags that were given to us.
+		file.Close()
```

Comment:
I'm a bit confused by this and what this is doing. Can you explain? Also, AFAIK, I don't think the flag/perm are implemented in the Windows go libraries, so I'm not sure it would have any effect.